client/components/domains/connect-domain-step/connect-domain-step-ownership-auth-code.jsx

```diff
@@ -0,0 +1,141 @@
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import ConnectDomainStepWrapper from './connect-domain-step-wrapper';
+import { modeType, stepsHeadingAdvanced, stepsHeadingSuggested, stepSlug } from './constants';
+
+import './style.scss';
+
+function ConnectDomainStepLogin( {
+	className,
+	defaultConnectHandler,
+	domain,
+	mode,
+	onConnect,
+	pageSlug,
+	progressStepList,
+	recordMappingButtonClickInUseYourDomain,
+	selectedSite,
+} ) {
+	const { __ } = useI18n();
+	const [ authCode, setAuthCode ] = useState( '' );
+	const [ authCodeCheckInProgress, setAuthCodeCheckInProgress ] = useState( false );
+	const [ connectInProgress, setConnectInProgress ] = useState( false );
+	const [ authCodeError, setAuthCodeError ] = useState( null );
+
+	const handleConnect = () => {
+		recordMappingButtonClickInUseYourDomain( domain );
+		setConnectInProgress( true );
+
+		const connectHandler = onConnect ?? defaultConnectHandler;
+		connectHandler( { domain, selectedSite }, () => setConnectInProgress( false ) );
+	};
+
+	const onAuthCodeChange = ( event ) => {
+		setAuthCode( event?.target?.value );
+		setAuthCodeError( null );
+	};
+
+	const checkAuthCode = () => {
+		if ( 0 === authCode.length ) {
+			setAuthCodeError( __( 'Please enter an authorization code.' ) );
+			return;
+		}
+
+		setAuthCodeCheckInProgress( true );
+		setAuthCodeError( null );
+
+		wpcom
+			.domains()
+			.authCodeCheck( domain, authCode )
+			.then( ( result ) => {
+				if ( true === result.success ) {
+					handleConnect();
+				} else {
+					setAuthCodeError(
+						__( 'Sorry, that authorization code is incorrect. Please try again.' )
+					);
+				}
+			} )
+			.catch( ( error ) => setAuthCodeError( error.message ) )
+			.finally( () => setAuthCodeCheckInProgress( false ) );
+	};
+
+	const stepContent = (
+		<div className={ className + '__login' }>
+			<p className={ className + '__text' }>
+				{ __(
+					'We will use your domain authorization code to verify that you are the domain owner.'
+				) }
+			</p>
+			<p className={ className + '__text' }>
+				{ __(
+					'A domain authorization code is a unique code linked only to your domain, it might also be called a secret code, auth code, or EPP code. You can usually find this in your domain settings page.'
+				) }
+			</p>
+			<div className={ className + '__authorization-code' }>
+				<Input
+					placeholder={ __( 'Enter your authorization code' ) }
+					onChange={ onAuthCodeChange }
+					value={ authCode }
+					isError={ !! authCodeError }
+					errorMessage={ authCodeError }
+				/>
+			</div>
+			<p className={ className + '__text' }>
+				{ __(
+					"Once you've entered the authorization code, click on the button below to proceed."
+				) }
+			</p>
+			<Button
+				busy={ authCodeCheckInProgress || connectInProgress }
+				disabled={ authCodeCheckInProgress || connectInProgress }
+				onClick={ checkAuthCode }
+				primary
+			>
+				{ __( 'Check authorization code' ) }
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Check my authorization code' )` *ES Score: 9*